skribenten-web/frontend/src/routes/saksnummer_/$saksId/brevvelger/$templateId/-components/endreMottaker/EndreMottaker.tsx

```diff
@@ -0,0 +1,324 @@
+
+import { finnSamhandler } from "~/api/skribenten-api-endpoints";
+import type { Adresse, FinnSamhandlerRequestDto, FinnSamhandlerResponseDto } from "~/types/apiTypes";
+import type { Nullable } from "~/types/Nullable";
+
+import { Route } from "../../route";
+import BekreftAvbrytelse from "./BekreftAvbrytelse";
+import type { EndreMottakerModalTabs, ManuellAdresseUtfyllingFormData } from "./EndreMottakerUtils";
+import {
+  type CombinedFormData,
+  createSamhandlerValidationSchema,
+  type FinnSamhandlerFormData,
+  InnOgUtland,
+  Søketype,
+} from "./EndreMottakerUtils";
+import HentOgVisSamhandlerAdresse from "./HentOgVisSamhandlerAdresse";
+import OppsummeringAvValgtMottaker from "./OppsummeringAvValgtMottaker";
+import SøkOgVelgSamhandlerForm from "./SøkOgVelgSamhandlerForm";
+import UtfyllingAvManuellAdresseForm from "./UtfyllingAvManuellAdresseForm";
+
+const EndreMottaker = (properties: {
+  //kan være undefined fordi vi ikke kan gjøre noe manuellAdresse enda
+  onManuellAdresseBekreft?: (a: Nullable<Adresse>) => void;
+}) => {
+  const [modalÅpen, setModalÅpen] = useState<boolean>(false);
+  const navigate = useNavigate({ from: Route.fullPath });
+
+  return (
+    <div>
+      {modalÅpen && (
+        <EndreMottakerModal
+          onBekreftNyMottaker={(bekreftetMottaker) => {
+            setModalÅpen(false);
+            //hvis mottaker er en string, betyr det at det er en samhandler
+            if (typeof bekreftetMottaker === "string") {
+              properties.onManuellAdresseBekreft?.(null);
+              navigate({
+                search: (s) => ({ ...s, idTSSEkstern: bekreftetMottaker }),
+                replace: true,
+              });
+            } else {
+              //hvis mottaker er en adresse, betyr det at det er en manuell adresse
+              properties.onManuellAdresseBekreft?.(bekreftetMottaker);
+              navigate({
+                search: (s) => ({ ...s, idTSSEkstern: undefined }),
+                replace: true,
+              });
+            }
+          }}
+          onClose={() => setModalÅpen(false)}
+          åpen={modalÅpen}
+        />
+      )}
+      <Button
+        data-cy="toggle-endre-mottaker-modal"
+        onClick={() => setModalÅpen(true)}
+        size="small"
+        type="button"
+        variant="secondary"
+      >
+        Endre mottaker
+      </Button>
+    </div>
+  );
+};
+
+export default EndreMottaker;
+
+/*
+  litt om hvorfor denne er laget som den er:
+  Vi har 1 modal, som skal deles opp i 3 tabs. 2 av dem er velgbare, mens 1 tab (oppsummering) er intern.
+  Modalen-bodyen skal reflektere 3 tilstander - Valg av mottaker, oppsummering, og avbrytelse
+  Oppsummering og avbyrtelse skal 'fjerne' alt det andre, derfor kreves det at modal har en state som vet hva den skal rendre
+  Vi klarer også å holde på diverse tilstander mye enklere, som f.eks sammhandler søk dersom dem navigerer seg tilbake fra oppsummering
+*/
+const EndreMottakerModal = (properties: {
```

Comment:
Er det mulig at form og validering blir mindre kompleks om vi ikke har samme form for alle tre mulighetene?

Reply:
yep - man kunne ha splittet det enda lenger enn bare det som er på tabsene. Derimot så vil man komme til samme problemstilling med håndtering av de ulike skisserte casene i modal-bodyen. 

Man vil da komme seg inn i et stort callback-helvette, som kan virke ganske uoversiktlig til en viss grad? Hver at disse mindre formene må nok ha 4(?) ish callbacks hver. Modal-bodyen må da håndtere 12+ callbacks. 

Jeg følte at det som er gjort nå, er generellt mindre smertefullt, enn alternativet